metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java

```diff
@@ -258,24 +331,38 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
     }
   }
 
-  private static class ArchiveEntryImpl implements ArchiveEntry {
+  private record ArchiveEntryImpl(String relativeFilePath, ByteArrayInputStream entryContent)
+      implements ArchiveEntry {
+
+    @Override
+    public String getHarvestingIdentifier() {
+      return relativeFilePath;
+    }
 
-    final String entryName;
-    final ByteArrayInputStream entryContent;
+    @Override
+    public void writeContent(OutputStream outputStream) throws IOException {
+       IOUtils.copy(entryContent, outputStream);
+    }
 
-    public ArchiveEntryImpl(String entryName, ByteArrayInputStream entryContent) {
-      this.entryName = entryName;
-      this.entryContent = entryContent;
+    @Override
+    public ByteArrayInputStream getContent() {
+      return entryContent;
     }
 
     @Override
-    public String getEntryName() {
-      return entryName;
+    public boolean isDeleted() {
+      return false;
```

Comment:
It always returns false, is this intentional?

Reply:
Yes. Other protocols (such as OAI) sometimes return records or record IDs that are deleted at source (with a flag saying that it is deleted). However, this is not the case here. In a zip file we can't make that distinction (we don't have such a flag) and we assume that all records are active and to be harvested.